File: lab3/Lab03_1/app/src/main/java/vn/edu/uit/lab03_1/DetailActivity.java
package vn.edu.uit.lab03_1;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class DetailActivity extends AppCompatActivity {

    private DatabaseHandler dbHandler;
    private Student student;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        dbHandler = new DatabaseHandler(this);

        final EditText etName = findViewById(R.id.et_detail_name);
        final EditText etStudentId = findViewById(R.id.et_detail_student_id);
        final EditText etClass = findViewById(R.id.et_detail_class);
        Button btnSave = findViewById(R.id.btn_save);
        Button btnCancel = findViewById(R.id.btn_cancel);

        student = (Student) getIntent().getSerializableExtra("student");

        if (student != null) {
            etName.setText(student.getName());
            etStudentId.setText(student.getStudentId());
            etClass.setText(student.getClassName());
        }

        btnSave.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = etName.getText().toString().trim();
                String studentId = etStudentId.getText().toString().trim();
                String className = etClass.getText().toString().trim();

                if (TextUtils.isEmpty(name) || TextUtils.isEmpty(studentId) || TextUtils.isEmpty(className)) {
                    Toast.makeText(DetailActivity.this, "Vui lòng nhập đầy đủ thông tin", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (student != null) {
                    student.setName(name);
                    student.setStudentId(studentId);
                    student.setClassName(className);
                    dbHandler.updateStudent(student);
                    Toast.makeText(DetailActivity.this, "Đã lưu thay đổi", Toast.LENGTH_SHORT).show();
                    setResult(RESULT_OK);
                }
                finish();
            }
        });

        btnCancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }
}

File: lab3/lab3_2/app/src/main/java/com/example/lab3_2/DetailActivity.kt
package com.example.lab3_2

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.appbar.MaterialToolbar

class DetailActivity : AppCompatActivity() {

    private lateinit var tvName: TextView
    private lateinit var tvMssv: TextView
    private lateinit var tvLop: TextView
    private lateinit var tvDiem: TextView
    private lateinit var btnEdit: Button
    private lateinit var btnDelete: Button
    private lateinit var toolbar: MaterialToolbar
    private lateinit var db: StudentDB

    private var studentId: Int = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        tvName = findViewById(R.id.tvName)
        tvMssv = findViewById(R.id.tvMssv)
        tvLop = findViewById(R.id.tvLop)
        tvDiem = findViewById(R.id.tvDiem)
        btnEdit = findViewById(R.id.btnEdit)
        btnDelete = findViewById(R.id.btnDelete)
        toolbar = findViewById(R.id.toolbar)
        db = StudentDB(this)

        toolbar.setNavigationOnClickListener {
            finish()
        }

        studentId = intent.getIntExtra("student_id", -1)

        if (studentId != -1) {
            val student = db.getStudentById(studentId)
            student?.let {
                tvName.text = it.name
                tvMssv.text = it.mssv
                tvLop.text = it.lop
                tvDiem.text = String.format("%.1f", it.diem)
            }
        }

        btnEdit.setOnClickListener {
            val intent = Intent(this, AddEditActivity::class.java)
            intent.putExtra("student_id", studentId)
            startActivity(intent)
            finish()
        }

        btnDelete.setOnClickListener {
            AlertDialog.Builder(this)
                .setTitle("Xác nhận")
                .setMessage("Bạn có muốn xóa sinh viên này?")
                .setPositiveButton("Xóa") { _, _ ->
                    db.deleteStudent(studentId)
                    Toast.makeText(this, "Đã xóa", Toast.LENGTH_SHORT).show()
                    startActivity(Intent(this, MainActivity::class.java))
                    finish()
                }
                .setNegativeButton("Hủy", null)
                .show()
        }
    }

    override fun onResume() {
        super.onResume()
        val student = db.getStudentById(studentId)
        student?.let {
            tvName.text = it.name
            tvMssv.text = it.mssv
            tvLop.text = it.lop
            tvDiem.text = String.format("%.1f", it.diem)
        }
    }
}
